pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -690,9 +692,51 @@ func (p *criStatsProvider) cleanupOutdatedCaches() {
 	}
 }
 
-// removeTerminatedContainer returns the specified container but with
-// the stats of the terminated containers removed.
-func removeTerminatedContainer(containers []*runtimeapi.Container) []*runtimeapi.Container {
+// removeTerminatedPods returns pods with terminated ones removed.
+// It only removes a terminated pod when there is a running instance
```

Comment:
What does this mean for the stats kubelet reports? If there are no running instances, kubelet would report the stats of the terminated containers?

Reply:
The terminated containers and terminated pods, I think.

And we've made sure that terminated containers will have required fields set, which we had issue before. This is still slightly different with cadvisor based summary api I guess.